decision_service/dsclientlib/main.cc

```diff
@@ -0,0 +1,84 @@
+#include "ds_driver.h"
+#include "ds_config_utility.h"
+
+#include <vector>
+#include <iostream>
+
+using namespace decision_service;
+using namespace decision_service::utility;
+using namespace decision_service::utility::config;
+
+config_collection load_config();
+void display_response(const ranking_response&);
+
+void error_handler(const api_status& error, void* user_context)
+{
+  std::cout << "An error happened in the background thread. Code:" << error.get_error_code() 
+    << " Details:" << error.get_error_msg() << std::endl;
+}
+
+int main()
+{
+  const auto config = init_from_json(R"({"eventhub_host":"localhost:8080"})");
```

Comment:
I'd suggest we walk through this carefully.

I'm a bit worried that initialization from json is adding work making use of the interface more difficult.

Reply:
Design principle was the following:

1) Initialize ds-clientlib using a name/value pair collection that has no knowledge of how configuration is stored.

2) The name/value pair collection can be populated in place (unit tests) or from other means e.g. a database table, csv config file, or a json config file etc.

3) We will supply utility methods so the current json config file stored in the user's app can be parsed and used to poplulate the config object.



---
In reply to: [187117181](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187117181) [](ancestors = 187117181)